rspace/src/main/scala/coop/rchain/rspace/merger/MergingLogic.scala

```diff
@@ -48,8 +48,8 @@ object MergingLogic {
 
       // Search for match
       def check(left: EventLogIndex, right: EventLogIndex): Iterator[Blake2b256Hash] = {
-        val p = producesActive(left)
-        val c = consumesActive(right)
+        val p = producesCreatedAndNotDestroyed(left)
+        val c = consumesCreatedAndNotDestroyed(right)
```

Comment:
This seems to be calculated multiple times, why not put this as part of event log index?

Reply:
when you combine two EventLogIndex, you need to recalculate. Because produce created in one of them might be destroyed in another. So in join index it won't be in this set, but it will be in one where it is created

Reply:
I'm not saying for joined index but for one operation of merging this is calculated multiple times on the same event log index. I see that `producesCreatedAndNotDestroyed` will be called two times on the same event log index.

https://github.com/rchain/rchain/blob/3cadf0e801a9978adc9f861eb601750fe55f9172/casper/src/main/scala/coop/rchain/casper/merging/DagMerger.scala#L38

https://github.com/rchain/rchain/blob/3cadf0e801a9978adc9f861eb601750fe55f9172/casper/src/main/scala/coop/rchain/casper/merging/DagMerger.scala#L58

I'm not sure how much this impacts performance but there is a lot of diffs.

Reply:
This is diff and append on sets, so I believe should be fast. We might cache if it will be slow